include/simdjson/generic/ondemand/document-inl.h

```diff
@@ -84,7 +84,12 @@ simdjson_really_inline document::operator std::string_view() noexcept(false) { r
 simdjson_really_inline document::operator raw_json_string() noexcept(false) { return get_raw_json_string(); }
 simdjson_really_inline document::operator bool() noexcept(false) { return get_bool(); }
 #endif
-
+simdjson_really_inline simdjson_result<size_t> document::count_elements() & noexcept {
+  auto a = get_array();
+  auto answer = a.count_elements();
+  if(answer.error() == SUCCESS) { iter.rewind(); }
```

Comment:
Why do we rewind here? It looks like we rewind inside count_elements itself ...

Reply:
Good point. It is one time too many.

Reply:
Oh wait. I was confused. This turns out to be critically important.